src/transformers/models/clip/modeling_flax_clip.py

```diff
@@ -0,0 +1,1087 @@
+class FlaxCLIPTextPreTrainedModel(FlaxPreTrainedModel):
+    config_class = CLIPTextConfig
+    module_class: nn.Module = None
+
+    def __init__(
+        self, config: CLIPTextConfig, input_shape=(1, 1), seed: int = 0, dtype: jnp.dtype = jnp.float32, **kwargs
+    ):
+        module = self.module_class(config=config, dtype=dtype, **kwargs)
+        super().__init__(config, module, input_shape=input_shape, seed=seed, dtype=dtype)
+
+    def init_weights(self, rng: jax.random.PRNGKey, input_shape: Tuple) -> FrozenDict:
+        # init input tensor
+        input_ids = jnp.zeros(input_shape, dtype="i4")
+        position_ids = jnp.broadcast_to(jnp.arange(jnp.atleast_2d(input_ids).shape[-1]), input_shape)
+        attention_mask = jnp.ones_like(input_ids)
+
+        params_rng, dropout_rng = jax.random.split(rng)
+        rngs = {"params": params_rng, "dropout": dropout_rng}
+
+        return self.module.init(rngs, input_ids, attention_mask, position_ids)["params"]
+
+    def __call__(
+        self,
+        input_ids,
+        attention_mask=None,
+        position_ids=None,
+        params: dict = None,
+        dropout_rng: jax.random.PRNGKey = None,
+        train: bool = False,
+        output_attentions: Optional[bool] = None,
+        output_hidden_states: Optional[bool] = None,
+        return_dict: Optional[bool] = None,
+    ):
+        output_attentions = output_attentions if output_attentions is not None else self.config.output_attentions
+        output_hidden_states = (
+            output_hidden_states if output_hidden_states is not None else self.config.output_hidden_states
+        )
+        return_dict = return_dict if return_dict is not None else self.config.return_dict
+
+        if output_attentions:
+            raise NotImplementedError(
+                "Currently attention scores cannot be returned. Please set `output_attentions` to False for now."
+            )
+
+        if position_ids is None:
+            position_ids = jnp.broadcast_to(jnp.arange(jnp.atleast_2d(input_ids).shape[-1]), input_ids.shape)
+
+        if attention_mask is None:
+            attention_mask = jnp.ones_like(input_ids)
+
+        # Handle any PRNG if needed
+        rngs = {}
+        if dropout_rng is not None:
+            rngs["dropout"] = dropout_rng
+
+        return self.module.apply(
+            {"params": params or self.params},
+            jnp.array(input_ids, dtype="i4"),
+            jnp.array(attention_mask, dtype="i4"),
+            jnp.array(position_ids, dtype="i4"),
+            not train,
+            output_attentions,
+            output_hidden_states,
+            return_dict,
+            rngs=rngs,
+        )
+
+
+class FlaxCLIPVisionPreTrainedModel(FlaxPreTrainedModel):
+    config_class = CLIPVisionConfig
+    module_class: nn.Module = None
+
+    def __init__(self, config: CLIPVisionConfig, seed: int = 0, dtype: jnp.dtype = jnp.float32, **kwargs):
+        input_shape = (1, config.image_size, config.image_size, 3)
```

Comment:
using hardcoded shape here, because the `conv` layer's weight dim depend on the `image_size` and we need config to access the `image_size`

Reply:
How do the conv layer require the image size in flax but not in PyTorch? In PyTorch it's only used for the Embeddings.

Reply:
Aah, sorry, I said it the wrong way.

What I mean to say is, in flax we init the model by doing a forward pass so we need to pass an example, which is created using this `input_shape`. Here the `num_patches` depend on the `image_size` and `patch_size`, and `num_patches` is used to init the embedding layer. 

The default `input_shape` is the shape used by pre-trained model which would be (224, 224, 3) but in our tests, we use different `image_size` but we don't pass `input_shape` in tests, so in this case, it results in in-correct `num_patches` and there is shape mismatch when adding the embeddings because `input_shape` and current `image_size` in the test are different.

Maybe we should pass `input_shape` when initing the model in tests.
cc @patrickvonplaten

Reply:
Aaaaah! I understand! Then I think you should be using an init argument for that, that defaults to `config.image_size`, so it's easy to change.

Though all config arguments are easy to change so it works fine as it is too. Happy to see what @patrickvonplaten and @LysandreJik think is best here!

Reply:
Agree with @sgugger, think we should provide an `input_shape` argument that defaults to `(1, config.image_size, config.image_size, 3)` if `None`. We also provide an `input_sahpe` to FlaxBert's `__init__(...)` function